src/llmq/quorums_instantsend.h

```diff
@@ -52,23 +52,100 @@ class CInstantSendDb
+     */
     void RemoveInstantSendLockMined(const uint256& hash, int nHeight);
+    /**
+     * Marks an InstantSend Lock as archived.
+     * @param batch Object used to batch many calls together
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height that the transaction was included at
+     */
     void WriteInstantSendLockArchived(CDBBatch& batch, const uint256& hash, int nHeight);
+    /**
+     * Archives and deletes all IS Locks which were mined into a block before nUntilHeight
+     * @param nUntilHeight Removes all IS Locks confirmed up until nUntilHeight
+     * @return returns an unordered_map of the hash of the IS Locks and a pointer object to the IS Locks for all IS Locks which were removed
+     */
     std::unordered_map<uint256, CInstantSendLockPtr> RemoveConfirmedInstantSendLocks(int nUntilHeight);
+    /**
+     * Removes IS Locks from the archive if the tx was confirmed 100 blocks before nUntilHeight
+     * @param nUntilHeight the height from which to base the remove of archive IS Locks
+     */
     void RemoveArchivedInstantSendLocks(int nUntilHeight);
+    /**
+     * Checks if the archive contains a certain IS Lock Hash
+     * @param islockHash The hash to check for
+     * @return Returns true if that IS Lock hash is present in the archive, otherwise returns false
+     */
     bool HasArchivedInstantSendLock(const uint256& islockHash);
+    /**
+     * Gets the number of IS Locks which have not been confirmed by a block
+     * @return size_t value of the number of IS Locks not confirmed by a block
+     */
     size_t GetInstantSendLockCount();
 
+    /**
+     * Gets a pointer to the IS Lock based on the hash
+     * @param hash The hash of the IS Lock
+     * @return A Pointer object to the IS Lock, returns nullptr if it doesn't exsist
+     */
     CInstantSendLockPtr GetInstantSendLockByHash(const uint256& hash);
+    /**
+     * Gets an IS Lock hash based on the txid the IS Lock is for
+     * @param txid The txid which is being searched for
+     * @return Returns the hash the IS Lock of the specified txid, returns uint256() if it doesn't exsist
+     */
     uint256 GetInstantSendLockHashByTxid(const uint256& txid);
+    /**
+     * Gets an IS Lock pointer from the txid given
+     * @param txid The txid for which the IS Lock Pointer is being returned
+     * @return Returns the IS Lock Pointer assosiated with the txid, returns nullptr if it doesn't exsist
+     */
     CInstantSendLockPtr GetInstantSendLockByTxid(const uint256& txid);
+    /**
+     * Gets an IS Lock pointer from an input given
+     * @param outpoint Since all inputs are really just outpoints that are being spent
+     * @return IS Lock Pointer assosiated with that input.
+     */
     CInstantSendLockPtr GetInstantSendLockByInput(const COutPoint& outpoint);
 
+    /**
+     * Gets a vector of IS Lock hashes of the IS Locks which rely on or are children of the parent IS Lock
```

Comment:
Same in regard to vector. This method returns "multiple instant send locks" and the fact that a vector is used is an implementation detail.

Reply:
Generally, you should try to document the purpose/meaning of a method and not implementation details, especially as everyone should be able to get the implementation details just by looking at the method declaration...if someone is not able to figure out the details based on that...he should close the IDE and go back to reading some tutorials or books...or stop being a coder :P

Reply:
Ah ... and I just read this, after writing what I did above. 🙈 😇